### DISCUSSION TITLE
Feature Requests 2022-06-20

### DISCUSSION THREAD

**Discussion by hamish-om** [gwleuverink/clickup-time-tracker#22] [Ideas]
Hi there, 

I have recently come across this and this and I'm finding it extremely helpful, so thanks!

I have a couple of ideas that could further improve usability:

- Ability to select billable or not

- Ability to edit description of time logged

- Show breadcrumbs (i.e. show what list and task the task/subtask belongs to)

- Timer functionality
- - - Enables you to simply put a description and not have to select a task before starting it (Easy and fast to use when you need to quickly start a timer)
- - - Enables you to pause the timer and not have to immediately assign a task (Easy and fast to use when you need to quickly start another new timer)
- - - Keeps timer data until the time log is committed (enables mass time logging at the end of day/end of week)

- Favourite tasks bar: separate to CU, enables users to have a list of frequent tasks they routinely log time for.
- - - I can sort of get around this by cmd-d'ing old tasks, but when new weeks start, or if the description is going to be different, the cmd-d feature gets a little cumbersome (still easier than logging it in ClickUp, so this is definitely a "nice to have" problem :) 

- Ability to connect with Google Calendar and log a G-calendar meeting as a clickup task or subtask (would enable meetings to be logged extremely quickly

- Ability to search subtasks and nested subtasks (this is probably a longshot given the amount of data that would need to be pulled)
- - -  If not possible to pull subtasks, As a compromise: 
- - - - - - ability to show subtasks that you are assigned to
- - - - - -  Have a "Tasks & Subtasks sorted by due date" as a side panel so people can see their priorities and log time from it

Some of these might not be possible, but thought i'd ask. Thanks for all your hard work on this!

**Comment by gwleuverink** [comment]
Hi, Glad it helps you!

Sorry about the slow response. Some things in my personal life kept me busy (and will for the next couple of months)
But rest assured I'm planning to add a lot of the things on your list. About the rest; I think they are pretty cool ideas, some quite doable, others a bit out of scope for the project. This because I have to maintain this in the future too, so I'm a bit careful with the workload & maintenance overhead it adds. 

**Quick wins on the shortlist:**
* Editing entries (description & billable)
* Add billable checkbox
* Ability to favorite tasks so they are sorted on top
* Ability to search subtasks and nested subtasks (need to check feasability) We already pull all tasks locally because the API doesn't allow me to fuzzy search tasks
* There seems to be some issue tracking time on subtasks

**Bigger additions:**
* Ability to see tracked time of other users in your organization (only when adding an admin token)
   _Need to make some changes to how things work under the hood & in the UI (like selecting other users) in order to make this work_
* Favourite tasks bar: separate to CU
   _I like this idea. I want to add favorite tasks to the top of the task dropdown first and then later add something like this_

**No plans to add at this time:**
* Ability to connect with Google Calendar  - simply because everyone's needs differ, some use Apple calendar, others Office 365. I can't support x different providers by myself with the little time I have. Though I do think this would be helpfull.
  _I believe there is a Google calendar integration for clickup, though I havent used it. Maybe you can sync clickup with your calendar & track it that way?_
* Show breadcrumbs - Not sure if the API gives me this info without making additional calls (I prefer to keep it as lightweight as possible). Need to check this though
* Timer functionality - personally I don't like working this way, but if you do, there are a couple of options that might suit your need, like the clickup browser extension or maybe a service like Toggl?
* Keeps timer data until the time log is committed - the beautiful thing about this app is that it delegates most logic to the clickup API. It is primarily a presentation layer on top. Though I agree this would be nice, it is a significant addition to do right

Hope this gives a more clear idea of my plans for this app. If you have any more suggestions or like to discuss ideas I'm happy to hear your input 👍🏻
